include/morphio/section_base.h

```diff
@@ -98,6 +83,16 @@ SectionBase<T>::SectionBase(uint32_t id, const std::shared_ptr<Property::Propert
     }
 }
 
+template <typename T>
+inline bool SectionBase<T>::operator==(const SectionBase& other) const noexcept {
+    return other.id_ == id_ && other.properties_ == properties_;
+}
+
+template <typename T>
+inline bool SectionBase<T>::operator!=(const SectionBase& other) const noexcept {
+    return !(*this == other);
+}
+
```

Comment:
Just for our sanity, wouldn't it be better to also declare them on the binds side?
```
pybind11_class.def(pybind11::self == pybind11::self);
pybind11_class.def(pybind11::self != pybind11::self);
```
For both mut and immut?

Reply:
python is already kinda doing this, from what I can tell.  If we need it, let's do it in another PR

Reply:
okk